### ISSUE TITLE
Type errors attempting to use curveCatmullRom from d3-shapes with <Area>

### ISSUE THREAD

**Issue by timtucker** [recharts/recharts#2499]
- [X] I have searched the [issues](https://github.com/recharts/recharts/issues) of this repository and believe that this is not a duplicate.

### Reproduction link
[![Edit on CodeSandbox](https://codesandbox.io/static/img/play-codesandbox.svg)](https://codesandbox.io/s/recharts-issue-template-forked-m7mlg?file=/src/index.tsx)

### Steps to reproduce
Create an <ComposedChart> with an <Area> and a <Line>
import curveCatmullRom from d3-shapes
attempt to use curveCatnullRom as the type for the <Area> and the <Line>

### What is expected?
Line and area are displayed with a catmull rom curve
No errors / warnings

### What is actually happening?
Getting a failure to compile locally and warnings when hovering over "type" in the sandbox when trying to use specify the curve for the <Area>
<Line> doesn't seem to have any issues.

Warning / failure to compile:
```
<Area type={curveCatmullRom} .../>
```
Produces the following error:
```
 Type 'CurveCatmullRomFactory' is not assignable to type '(string & CurveType) | undefined'.
      Type 'CurveCatmullRomFactory' is not assignable to type 'string & CurveFactory'.
        Type 'CurveCatmullRomFactory' is not assignable to type 'string'.
```

| Environment | Info |
|---|---|
| Recharts | v2.0.8 |
| React | 17 |
| System | Windows 10 |
| Browser | Chrome |

**Comment by timtucker** [comment]
Just realized this may be the same issue as #2471

**Comment by timtucker** [comment]
Looks like this is fixed now that #2471 is fixed
